src/runtime_functions.js

```diff
@@ -94,9 +94,30 @@ function convertJsFunctionToWasm(func, sig) {
 
 var freeTableIndexes = [];
 
+#if WASM_BACKEND
+// Weak map of functions in the table to their indexes, created on first use.
```

Comment:
Why wasm backend only?  if this behaviour change is going to be made why of make it apply to fastcomp too?

Reply:
Well, I only really understand how function pointers work in upstream, after spending the day on that. So this might work on fastcomp, but it's risky - as it's deprecated I figured safest to avoid any changes?

Reply:
But all this change is going is duplicating the results of adding functions to the wasm table.    Surly this will trivially work in both backends and if the test passes.. why not just remove the ifdef.   At the very least there is a readability cost to adding more ifdefs.

Reply:
If you think the risk is worth it I can remove the ifdef, sure.